src/extension.ts

```diff
@@ -27,23 +27,28 @@ export async function activate(context: vscode.ExtensionContext) {
   await clangdContext.activate(context.globalStoragePath, outputChannel,
                                context.workspaceState);
 
-  setInterval(function() {
-    const cppTools = vscode.extensions.getExtension('ms-vscode.cpptools');
-    if (cppTools && cppTools.isActive) {
-      const cppToolsConfiguration = vscode.workspace.getConfiguration('C_Cpp');
-      const cppToolsEnabled = cppToolsConfiguration.get('intelliSenseEngine');
-      if (cppToolsEnabled !== 'Disabled') {
-        vscode.window
-            .showWarningMessage(
-                'You have both the Microsoft C++ (cpptools) extension and ' +
-                    'clangd extension enabled. The Microsoft IntelliSense features ' +
-                    'conflict with clangd\'s code completion, diagnostics etc.',
-                'Disable IntelliSense')
-            .then(_ => {
-              cppToolsConfiguration.update('intelliSenseEngine', 'Disabled',
-                                           vscode.ConfigurationTarget.Global);
-            });
+  const shouldCheck = vscode.workspace
+    .getConfiguration("clangd")
+    .get("detectAndDisableExtensionConflicts");
+  if (shouldCheck) {
+    setInterval(function() {
+      const cppTools = vscode.extensions.getExtension('ms-vscode.cpptools');
+      if (cppTools && cppTools.isActive) {
+        const cppToolsConfiguration = vscode.workspace.getConfiguration('C_Cpp');
+        const cppToolsEnabled = cppToolsConfiguration.get('intelliSenseEngine');
+        if (cppToolsEnabled !== 'Disabled') {
+          vscode.window
+              .showWarningMessage(
+                  'You have both the Microsoft C++ (cpptools) extension and ' +
+                      'clangd extension enabled. The Microsoft IntelliSense features ' +
+                      'conflict with clangd\'s code completion, diagnostics etc.',
+                  'Disable IntelliSense')
```

Comment:
Nit: I think the perfect UX/UI would be adding a button `Don't show this warning`.

Something like 

```typescript
vscode.window.showWarningMessage('', ['Disable IntelliSense', 'Don\'t show this warning']).
    then(selection => {
        if (selection = 'Don\'t show this warning') {
            /* update clangd.detectAndDisableExtensionConflicts similar to what we
               have with C_Cpp.intelliSenseEngine right now */
        }
    }
```